### PR TITLE
Case study models

### PR DESCRIPTION
**Pull request by gnanakeethan**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request
Implements Models for PR

### CODE REVIEW

**Review by jmcook1186 — APPROVED** [review]
lgtm
only thing to flag here is that in this current state `sci-e` is a hard requirement in every pipeline, because `sci` expects to find `energy`, and only `sci-e` adds `energy` to the impacts. Other models return `e-cpu` `e-net` or `e-mem`. Not blocking, but something to return to later.

File: src/lib/case-studies/aveva-model.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import {IImpactModelInterface} from '../interfaces';
+import {KeyValuePair} from '../../types/boavizta';
+
+export class EAvevaModel implements IImpactModelInterface {
+  // Defined for compatibility. Not used in TEADS.
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/aveva-model.ts:5]: teads in aveva model

File: src/lib/case-studies/aveva-model.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  name: string | undefined;
+  /**
+   * Defined for compatibility. Not used here.
+   */
+  authenticate(authParams: object): void {
+    this.authParams = authParams;
+  }
+
+  /**
+   *  Configures the TEADS Plugin for IEF
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/aveva-model.ts:17]: teads

File: src/lib/case-studies/aveva-model.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
...KG9ic2VydmF0aW9uOiBLZXlWYWx1ZVBhaXIpIHtcbiAgICBpZiAoISgnbWVtLXV0aWwnIGluIG9ic2VydmF0aW9uKSB8fCAhKCd0aW1lc3RhbXAnIGluIG9ic2VydmF0aW9uKSkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKFxuICAgICAgICAnUmVxdWlyZWQgUGFyYW1ldGVycyBkdXJhdGlvbixjcHUtdXRpbCx0aW1lc3RhbXAgbm90IHByb3ZpZGVkIGZvciBvYnNlcnZhdGlvbidcbiAgICAgICk7XG4gICAgfVxuICAgIGlmICh0aGlzLm1lbW9yeUFsbG9jYXRpb24gPT09IDApIHtcbiAgICAgIHRocm93IG5ldyBFcnJvcihcbiAgICAgICAgJ1JlcXVpcmVkIFBhcmFtZXRlcjogbWVtLWFsbG9jIG5vdCBwcm92aWRlZCBpbiBjb25maWd1cmUnXG4gICAgICApO1xuICAgIH1cbiAgICBpZiAodGhpcy5tZW1vcnlFbmVyZ3kgPT09IDApIHtcbiAgICAgIHRocm93IG5ldyBFcnJvcihcbiAgICAgICAgJ1JlcXVpcmVkIFBhcmFtZXRlcjogbWVtLWVuZXJneSBub3QgcHJvdmlkZWQgaW4gY29uZmlndXJlJ1xuICAgICAgKTtcbiAgICB9XG5cbiAgICBjb25zdCBtZW1fYWxsb2MgPSB0aGlzLm1lbW9yeUFsbG9jYXRpb247XG4gICAgLy8gICAgY29udmVydCBjcHUgdXNhZ2UgdG8gcGVyY2VudGFnZVxuICAgIGNvbnN0IG1lbV91dGlsID0gb2JzZXJ2YXRpb25bJ21lbS11dGlsJ107XG4gICAgaWYgKG1lbV91dGlsIDwgMCB8fCBtZW1fdXRpbCA+IDEwMCkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKCdjcHUgdXNhZ2UgbXVzdCBiZSBiZXR3ZWVuIDAgYW5kIDEwMCcpO1xuICAgIH1cblxuICAgIGNvbnN0IG1lbV9lbmVyZ3kgPSB0aGlzLm1lbW9yeUVuZXJneTtcblxuICAgIHJldHVybiAobWVtX2FsbG9jICogKG1lbV91dGlsIC8gMTAwKSAqIG1lbV9lbmVyZ3kpIC8gMTAwMDtcbiAgfVxufVxuIl19
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/aveva-model.ts:87]: what is this?

File: src/lib/case-studies/aveva.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
...ZGVsLmNhbGN1bGF0ZShbXG4gICAgICAgIHtcbiAgICAgICAgICBkdXJhdGlvbjogMzYwMCxcbiAgICAgICAgICAnbWVtLXV0aWwnOiAxMC4wLFxuICAgICAgICAgIHRpbWVzdGFtcDogJzIwMjEtMDEtMDFUMDA6MDA6MDBaJyxcbiAgICAgICAgfSxcbiAgICAgICAge1xuICAgICAgICAgIGR1cmF0aW9uOiAzNjAwLFxuICAgICAgICAgICdtZW0tdXRpbCc6IDUwLjAsXG4gICAgICAgICAgdGltZXN0YW1wOiAnMjAyMS0wMS0wMVQwMDowMDowMFonLFxuICAgICAgICB9LFxuICAgICAgICB7XG4gICAgICAgICAgZHVyYXRpb246IDM2MDAsXG4gICAgICAgICAgJ21lbS11dGlsJzogOTAuMCxcbiAgICAgICAgICB0aW1lc3RhbXA6ICcyMDIxLTAxLTAxVDAwOjAwOjAwWicsXG4gICAgICAgIH0sXG4gICAgICBdKVxuICAgICkucmVzb2x2ZXMudG9TdHJpY3RFcXVhbChbXG4gICAgICB7XG4gICAgICAgICdlLW1lbSc6IDAuMDAxMjE2MDAwMDAwMDAwMDAwMyxcbiAgICAgICAgZHVyYXRpb246IDM2MDAsXG4gICAgICAgICdtZW0tdXRpbCc6IDEwLjAsXG4gICAgICAgIHRpbWVzdGFtcDogJzIwMjEtMDEtMDFUMDA6MDA6MDBaJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgICdlLW1lbSc6IDAuMDA2MDgsXG4gICAgICAgIGR1cmF0aW9uOiAzNjAwLFxuICAgICAgICAnbWVtLXV0aWwnOiA1MC4wLFxuICAgICAgICB0aW1lc3RhbXA6ICcyMDIxLTAxLTAxVDAwOjAwOjAwWicsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICAnZS1tZW0nOiAwLjAxMDk0NCxcbiAgICAgICAgZHVyYXRpb246IDM2MDAsXG4gICAgICAgICdtZW0tdXRpbCc6IDkwLjAsXG4gICAgICAgIHRpbWVzdGFtcDogJzIwMjEtMDEtMDFUMDA6MDA6MDBaJyxcbiAgICAgIH0sXG4gICAgXSk7XG4gIH0pO1xufSk7XG4iXX0=
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/aveva.test.ts:30]: this one also

File: src/lib/case-studies/emem-model.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
+import {IImpactModelInterface} from '../interfaces';
+import {KeyValuePair} from '../../types/boavizta';
+
+export class EMemModel implements IImpactModelInterface {
+  // Defined for compatibility. Not used in TEADS.
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/emem-model.ts:5]: teads

File: src/lib/case-studies/emem-model.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  authParams: object | undefined;
+  // name of the data source
+  name: string | undefined;
+  // tdp of the chip being measured
+  memoryAllocation = 0;
+  // default power curve provided by the Teads Team
+  memoryEnergy = 0;
+  /**
+   * Defined for compatibility. Not used in TEADS.
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/emem-model.ts:14]: teads

File: src/lib/case-studies/emem-model.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+   */
+  authenticate(authParams: object): void {
+    this.authParams = authParams;
+  }
+
+  /**
+   *  Configures the TEADS Plugin for IEF
```
Comment (narekhovhannisyan) [inline src/lib/case-studies/emem-model.ts:21]: teads

File: src/util/models-universe.ts [hunk 8]
```diff
@@ -36,10 +45,25 @@ export class ModelsUniverse {
...
+        return SciModel;
+      case 'eshoppen-net':
+        return EshoppenNetModel;
+      case 'eshoppen-cpu':
+        return EshoppenCpuModel;
+      case 'eshoppen-mem':
+        return EshoppenMemModel;
+      case 'sci-accenture':
+        return SciAccentureModel;
+
```
Comment (narekhovhannisyan) [inline src/util/models-universe.ts:64]: redundant newline

**Comment by gnanakeethan** [comment]
@jmcook1186 I rebased. Please do not reverse merge the remote.

File: src/util/models-universe.ts [hunk 9]
```diff
@@ -13,6 +13,15 @@ import {
...
   InitalizedModels,
 } from '../types/models-universe';
+import {SciModel} from '../lib/sci';
+import {
+  EshoppenCpuModel,
+  EshoppenMemModel,
+  EshoppenNetModel,
+} from '../lib/case-studies/eshoppen-model';
+import {EMemModel} from '../lib/case-studies/emem-model';
+import {SciAccentureModel} from '../lib/case-studies/sci-accenture-model';
```
Comment (narekhovhannisyan) [inline src/util/models-universe.ts:23]: please have index file in case-studies and re-export all models from there

**Review by narekhovhannisyan — APPROVED** [review]

**Pull request merged**